8.3.60 incompatible with Python 3.8

### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.


### Ultralytics YOLO Component

_No response_

### Bug

Hi, it looks like the latest Ultralytics version (8.3.60) is not compatible with Python 3.8.
I get the following error on import:
```
.venv/lib/python3.8/site-packages/ultralytics/utils/__init__.py:1246: in update
    kwargs |= arg
E   TypeError: unsupported operand type(s) for |=: 'dict' and 'dict'
```

This is most likely due to the ruff 0.9.0 autoformat changes from #18622 

Union operators for dicts are only supported in Python 3.9+. Would it be possible to either revert this change or update the minimal Python version in pyproject.toml to 3.9?


### Environment

-

### Minimal Reproducible Example

-

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -165,7 +165,7 @@ jobs:
         torch: [latest]
         include:
           - os: ubuntu-latest
-            python-version: "3.9" # torch 1.8.0 requires python >=3.6, <=3.9
+            python-version: "3.8" # torch 1.8.0 requires python >=3.6, <=3.9
             torch: "1.8.0" # min torch version CI https://pypi.org/project/torchvision/
     steps:
       - uses: actions/checkout@v4

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -39,8 +39,7 @@ on:
 
 jobs:
   HUB:
-    # if: github.repository == 'ultralytics/ultralytics' && (github.event_name == 'schedule' || github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.hub == 'true'))
-    if: github.repository == 'ultralytics/ultralytics' && 'workflow_dispatch' && github.event.inputs.hub == 'true'
+    if: github.repository == 'ultralytics/ultralytics' && (github.event_name == 'schedule' || github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.hub == 'true'))
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false

--- a/ultralytics/__init__.py
+++ b/ultralytics/__init__.py
@@ -1,6 +1,6 @@
 # Ultralytics YOLO 🚀, AGPL-3.0 license
 
-__version__ = "8.3.60"
+__version__ = "8.3.61"
 
 import os
 

--- a/ultralytics/models/sam/model.py
+++ b/ultralytics/models/sam/model.py
@@ -106,7 +106,7 @@ class SAM(Model):
             ...     print(f"Detected {len(r.masks)} masks")
         """
         overrides = dict(conf=0.25, task="segment", mode="predict", imgsz=1024)
-        kwargs = overrides | kwargs
+        kwargs = {**overrides, **kwargs}
         prompts = dict(bboxes=bboxes, points=points, labels=labels)
         return super().predict(source, stream, prompts=prompts, **kwargs)
 

--- a/ultralytics/models/sam/modules/utils.py
+++ b/ultralytics/models/sam/modules/utils.py
@@ -54,7 +54,7 @@ def select_closest_cond_frames(frame_idx, cond_frame_outputs, max_cond_frame_num
             (t for t in cond_frame_outputs if t not in selected_outputs),
             key=lambda x: abs(x - frame_idx),
         )[:num_remain]
-        selected_outputs |= ((t, cond_frame_outputs[t]) for t in inds_remain)
+        selected_outputs.update((t, cond_frame_outputs[t]) for t in inds_remain)
         unselected_outputs = {t: v for t, v in cond_frame_outputs.items() if t not in selected_outputs}
 
     return selected_outputs, unselected_outputs

--- a/ultralytics/models/utils/loss.py
+++ b/ultralytics/models/utils/loss.py
@@ -243,8 +243,10 @@ class DETRLoss(nn.Module):
         if len(gt_bboxes):
             gt_scores[idx] = bbox_iou(pred_bboxes.detach(), gt_bboxes, xywh=True).squeeze(-1)
 
-        loss = {**self._get_loss_class(pred_scores, targets, gt_scores, len(gt_bboxes), postfix),
-        **self._get_loss_bbox(pred_bboxes, gt_bboxes, postfix)}
+        loss = {
+            **self._get_loss_class(pred_scores, targets, gt_scores, len(gt_bboxes), postfix),
+            **self._get_loss_bbox(pred_bboxes, gt_bboxes, postfix),
+        }
         # if masks is not None and gt_mask is not None:
         #     loss.update(self._get_loss_mask(masks, gt_mask, match_indices, postfix))
         return loss

--- a/ultralytics/models/utils/loss.py
+++ b/ultralytics/models/utils/loss.py
@@ -243,9 +243,8 @@ class DETRLoss(nn.Module):
         if len(gt_bboxes):
             gt_scores[idx] = bbox_iou(pred_bboxes.detach(), gt_bboxes, xywh=True).squeeze(-1)
 
-        loss = {}
-        loss |= self._get_loss_class(pred_scores, targets, gt_scores, len(gt_bboxes), postfix)
-        loss.update(self._get_loss_bbox(pred_bboxes, gt_bboxes, postfix))
+        loss = {**self._get_loss_class(pred_scores, targets, gt_scores, len(gt_bboxes), postfix),
+        **self._get_loss_bbox(pred_bboxes, gt_bboxes, postfix)}
         # if masks is not None and gt_mask is not None:
         #     loss.update(self._get_loss_mask(masks, gt_mask, match_indices, postfix))
         return loss

--- a/ultralytics/solutions/streamlit_inference.py
+++ b/ultralytics/solutions/streamlit_inference.py
@@ -62,7 +62,7 @@ class Inference:
         self.selected_ind = []  # List of selected classes for detection or tracking
         self.model = None  # Container for the loaded model instance
 
-        self.temp_dict = {"model": None} | kwargs
+        self.temp_dict = {"model": None, **kwargs}
         self.model_path = None  # Store model file name with path
         if self.temp_dict["model"] is not None:
             self.model_path = self.temp_dict["model"]

--- a/ultralytics/utils/__init__.py
+++ b/ultralytics/utils/__init__.py
@@ -1243,7 +1243,7 @@ class SettingsManager(JSONDict):
         """Updates settings, validating keys and types."""
         for arg in args:
             if isinstance(arg, dict):
-                kwargs |= arg
+                kwargs.update(arg)
         for k, v in kwargs.items():
             if k not in self.defaults:
                 raise KeyError(f"No Ultralytics setting '{k}'. {self.help_msg}")

